com.unity.perception/Runtime/GroundTruth/PerceptionCamera.cs

```diff
@@ -157,155 +162,82 @@ public bool RemovePersistentSensorData(string key)
-
-        void SetupVisualizationCamera(Camera cam)
-        {
-#if !UNITY_EDITOR && !DEVELOPMENT_BUILD
-            showVisualizations = false;
-#else
-
-            var visualizationAllowed = s_VisualizedPerceptionCamera == null;
-
-            if (!visualizationAllowed && showVisualizations)
-            {
-                Debug.LogWarning($"Currently only one PerceptionCamera may be visualized at a time. Disabling visualization on {gameObject.name}.");
-                showVisualizations = false;
-                return;
-            }
-            if (!showVisualizations)
-                return;
-
-            m_ShowingVisualizations = true;
-            s_VisualizedPerceptionCamera = this;
-
-            hudPanel = gameObject.AddComponent<HUDPanel>();
-            overlayPanel = gameObject.AddComponent<OverlayPanel>();
-            overlayPanel.perceptionCamera = this;
-#endif
-        }
-
-        void CheckForRendererFeature(ScriptableRenderContext context, Camera camera)
-        {
-            if (camera == m_AttachedCamera)
-            {
-#if URP_PRESENT
-                if (!m_GroundTruthRendererFeatureRun)
-                {
-                    Debug.LogError("GroundTruthRendererFeature must be present on the ScriptableRenderer associated with the camera. The ScriptableRenderer can be accessed through Edit -> Project Settings... -> Graphics -> Scriptable Render Pipeline Settings -> Renderer List.");
-                    enabled = false;
-                }
-#endif
-                RenderPipelineManager.endCameraRendering -= CheckForRendererFeature;
-            }
-        }
-        // Update is called once per frame
         void Update()
         {
             EnsureSensorRegistered();
             if (!SensorHandle.IsValid)
                 return;
 
-            bool anyVisualizing = false;
             foreach (var labeler in m_Labelers)
             {
                 if (!labeler.enabled)
                     continue;
 
                 if (!labeler.isInitialized)
-                {
                     labeler.Init(this);
-                }
 
                 labeler.InternalOnUpdate();
-                anyVisualizing |= labeler.InternalVisualizationEnabled;
             }
 
             // Currently there is an issue in the perception camera that causes the UI layer not to be visualized
-            // if we are utilizing async readback and we have to flip our captured image. We have created a jira
-            // issue for this (aisv-779) and have notified the engine team about this.
-            anyVisualizing = true;
-
+            // if we are utilizing async readback and we have to flip our captured image.
+            // We have created a jira issue for this (aisv-779) and have notified the engine team about this.
             if (m_ShowingVisualizations)
-                CaptureOptions.useAsyncReadbackIfSupported = !anyVisualizing;
-        }
-
-        private Vector2 scrollPosition;
```

Comment:
These large code movements make an already complex PR significantly harder to review. I realize I am also guilty of this on a fairly regular basis, but do be cautious about where you make these kinds of changes. Preferably they would be saved for a separate PR

Reply:
Good call-out, I jumped the gun by making a few too many changes to this file while poking around, and so I thought that I might as well go all in and get it done now instead of half here and then another half in another PR.